### PR TITLE
Tidy EFI defines / add defines for MegaSquirt and Lutan

### PR DESCRIPTION
**Pull request by peterbarker**
- canonicalises on AP_EFI.*
 - moves backend defines into config.h
 - NFC

```
Board              AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                      *      *           *       *                 *      *      *
Hitec-Airspeed     *                                                                     
KakuteH7-bdshot               *      *           *       *                 *      *      *
MatekF405                     *      *           *       *                 *      *      *
Pixhawk1-1M                   *      *           *       *                 *      *      *
f103-QiotekPeriph  *                                                                     
f303-Universal     *                                                                     
iomcu                                                          *                         
revo-mini                     *      *           *       *                 *      *      *
skyviper-v2450                                   *                                       
```

### CODE REVIEW

**Comment by peterbarker** [comment]
I've run this through the build options testing script and it's fine.

File: libraries/AP_EFI/AP_EFI_DroneCAN.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #include "AP_EFI.h"
 #include "AP_EFI_Backend.h"
 
-#if HAL_EFI_DRONECAN_ENABLED
+#if AP_EFI_DRONECAN_ENABLED
```
Comment (khancyr) [inline libraries/AP_EFI/AP_EFI_DroneCAN.h:6]: could probably move this up, and add the #include "AP_EFI_config.h" first to have the same pattern as other

**Review by khancyr — APPROVED** [review]
LGTM
Changes are just moving compile time verified stuff.
move define setting into the xxx_config.h file. same check as before. include order is correct

**Comment by peterbarker** [comment]
Ah.  I've just noted we're missing some prerequisites for the DroneCAN in the build_options.py file.  Darn.  Will fix those and test the build_options.py file

**Comment by peterbarker** [comment]
> Ah. I've just noted we're missing some prerequisites for the DroneCAN in the build_options.py file. Darn. Will fix those and test the build_options.py file

.... no I didn't, we don't have build_options entries for either of CAN or scripting ATM.

**Pull request merged**